### PR TITLE
REGR: None converted to NaN when enlarging Series

### LINKED ISSUE #48665: BUG: None value converted to NaN in Series
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
>>> import pandas as pd
>>> s = pd.Series(['a', 'b'])
>>> s
0    a
1    b
dtype: object
>>> s['a'] = None
>>> s
0      a
1      b
a    NaN
dtype: object
```

### Issue Description

In 1.5.0 there is a behaviour change, if you set a value on a Serie to None it is replaced with NaN.

Same thing with `s.loc['dd'] = None`

### Expected Behavior

It should stay None

### Installed Versions

<details>
>>> pd.show_versions()

INSTALLED VERSIONS
------------------
commit           : 9a607e2afa549a9295db09ad05df6572c1a85542
python           : 3.8.13.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 21.6.0
Version          : Darwin Kernel Version 21.6.0: Wed Aug 10 14:25:27 PDT 2022; root:xnu-8020.141.5~2/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : en_GB.UTF-8
LOCALE           : en_GB.UTF-8

pandas           : 1.6.0.dev0+165.g9a607e2afa
numpy            : 1.22.3
pytz             : 2021.3
dateutil         : 2.8.2
setuptools       : 56.0.0
pip              : 22.0.4
Cython           : 0.29.32
pytest           : 6.2.5
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : 1.1
pymysql          : None
psycopg2         : None
jinja2           : 3.0.3
IPython          : 7.16.1
pandas_datareader: None
bs4              : 4.11.1
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : 0.8.7
gcsfs            : None
matplotlib       : 3.5.1
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 3.0.0
pyreadstat       : None
pyxlsb           : None
s3fs             : 0.3.5
scipy            : 1.5.4
snappy           : None
sqlalchemy       : 1.4.32
tables           : None
tabulate         : 0.8.9
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None
tzdata           : None

</details>

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #48665 (Replace xxxx with the Github issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/series/indexing/test_setitem.py [hunk 1]
```diff
@@ -562,6 +562,14 @@ def test_setitem_enlarge_with_na(self, na, target_na, dtype, target_dtype, index
         expected = Series(expected_values, dtype=target_dtype)
         tm.assert_series_equal(ser, expected)
 
+    def test_setitem_enlargement_object_none(self):
+        # GH#48665
+        ser = Series(["a", "b"])
+        ser[3] = None
```
Comment (mroeschke) [inline pandas/tests/series/indexing/test_setitem.py:568]: Would it be good to use `nulls_fixture` to check that setting all null-like values result in object dtype?

Comment (phofl) [reply pandas/tests/series/indexing/test_setitem.py:568]: Don’t really know how to test that,
assert_frame_equal considers NaN equal to None for object dtypes, same for NaT and NaN

Comment (mroeschke) [reply pandas/tests/series/indexing/test_setitem.py:568]: Hmm could check that the `__getitem__` returns the `null_fixture` like you do below:

```
assert ser[3] is nulls_fixture
```

Comment (phofl) [reply pandas/tests/series/indexing/test_setitem.py:568]: That would probably work, yes

Comment (phofl) [reply pandas/tests/series/indexing/test_setitem.py]: Added

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @phofl

**Pull request merged**